lib/Conversion/FIRRTLToRTL/LowerToRTL.cpp

```diff
@@ -1696,58 +1694,64 @@ LogicalResult FIRRTLLowering::visitDecl(MemOp op) {
       continue;
     case MemOp::PortKind::Write: {
       SmallVector<WritePipeElement> writePipe;
-      writePipe.push_back({portWires["en"], portWires["addr"],
-                           portWires["mask"], portWires["data"]});
+      writePipe.push_back(
+          {portWires["en"], portWires["addr"], portWires["mask"],
+           portWires["data"], builder->create<sv::ReadInOutOp>(portWires["en"]),
+           builder->create<sv::ReadInOutOp>(portWires["addr"]),
+           builder->create<sv::ReadInOutOp>(portWires["mask"]),
+           builder->create<sv::ReadInOutOp>(portWires["data"])});
```

Comment:
You know C++ better than I do, but just make sure this is stable.

Reply:
you make a fine point.